App shouldn't crash if you type your password wrong

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.apereo/org.apereo.activities.HomePage_}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.List org.apereo.models.Layout.getFolders()' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2298)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360)
        at android.app.ActivityThread.access$800(ActivityThread.java:144)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:135)
        at android.app.ActivityThread.main(ActivityThread.java:5221)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.List org.apereo.models.Layout.getFolders()' on a null object reference
        at org.apereo.activities.HomePage.init(HomePage.java:79)
        at org.apereo.activities.HomePage_.onViewChanged(HomePage_.java:92)
        at org.androidannotations.api.view.OnViewChangedNotifier.notifyViewChanged(OnViewChangedNotifier.java:41)
        at org.apereo.activities.HomePage_.setContentView(HomePage_.java:57)
        at org.apereo.activities.HomePage_.onCreate(HomePage_.java:44)
        at android.app.Activity.performCreate(Activity.java:5933)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2251)
        ... 10 more
```

--- a/app/src/main/java/org/apereo/activities/BaseActivity.java
+++ b/app/src/main/java/org/apereo/activities/BaseActivity.java
@@ -39,10 +39,12 @@ public class BaseActivity extends Activity {
         super.onDestroy();
     }
 
+    @UiThread
     public void showShortToast(String msg) {
         Toast.makeText(this, msg, Toast.LENGTH_SHORT).show();
     }
 
+    @UiThread
     public void showLongToast(String msg) {
         Toast.makeText(this, msg, Toast.LENGTH_LONG).show();
     }

--- a/app/src/main/java/org/apereo/activities/LoginActivity.java
+++ b/app/src/main/java/org/apereo/activities/LoginActivity.java
@@ -86,7 +86,6 @@ public class LoginActivity extends BaseActivity {
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         getActionBar().setDisplayHomeAsUpEnabled(true);
-
     }
 
     @AfterViews
@@ -170,8 +169,13 @@ public class LoginActivity extends BaseActivity {
             }
 
             @Override
-            public void onError(Exception e, String responseBody) {
-                Logger.e(TAG, e.getMessage(), e);
+            public void onError(final Exception e, final String responseBody) {
+                dismissSpinner();
+                if (e != null && !e.getMessage().isEmpty()) {
+                    showLongToast(e.getMessage());
+                } else if (!responseBody.isEmpty()) {
+                    showLongToast(responseBody);
+                }
             }
         });
     }

--- a/app/src/main/java/org/apereo/activities/PortletWebViewActivity.java
+++ b/app/src/main/java/org/apereo/activities/PortletWebViewActivity.java
@@ -27,7 +27,6 @@ import org.apereo.R;
 import org.apereo.adapters.FolderListAdapter;
 import org.apereo.models.Folder;
 import org.apereo.utils.LayoutManager;
-import org.apereo.utils.Logger;
 
 import java.util.List;
 
@@ -118,7 +117,6 @@ public class PortletWebViewActivity extends BaseActivity implements AdapterView.
         //TODO find better way to do this
         url = url.replaceAll("/f/welcome","");
 
-        Logger.d(TAG, url);
         webView.loadUrl(url);
     }
 

--- a/app/src/main/java/org/apereo/utils/CasClient.java
+++ b/app/src/main/java/org/apereo/utils/CasClient.java
@@ -55,11 +55,15 @@ public class CasClient {
             // Perform the CAS authentication dance.
             List<NameValuePair> postData = getAndParsePostData(username, password);
             String stLocation = sendPostForServiceTicketLocation(postData);
-            String st = sendPostForServiceTicket(stLocation);
-            validateServiceTicket(st);
-            syncCookies();
+            if (stLocation != null) {
+                String st = sendPostForServiceTicket(stLocation);
+                validateServiceTicket(st);
+                syncCookies();
 
-            callback.onSuccess(null);
+                callback.onSuccess(null);
+            } else {
+                callback.onError(null, "Username or password is incorrect.");
+            }
         } catch (MalformedURLException e) {
             callback.onError(e, null);
         } catch (IOException e) {
@@ -122,9 +126,11 @@ public class CasClient {
 
         // Service ticket created.
         String serviceTicketLocation = postConnection.getHeaderField("Location");
-        serviceTicketLocation = serviceTicketLocation.replace("http", "https");
 
-        tgt = serviceTicketLocation.split("tickets/")[1];
+        if (serviceTicketLocation != null) {
+            serviceTicketLocation = serviceTicketLocation.replace("http", "https");
+            tgt = serviceTicketLocation.split("tickets/")[1];
+        }
 
         return serviceTicketLocation;
     }

--- a/app/src/main/res/values/strings.xml
+++ b/app/src/main/res/values/strings.xml
@@ -11,8 +11,8 @@
     <string name="login">Log In</string>
     <string name="logout">Log Out</string>
     <string name="error_title">Error</string>
-    <string name="form_error">Please enter a username and password</string>
-    <string name="information_error">Please check your username and password</string>
+    <string name="form_error">Please enter a username and password.</string>
+    <string name="information_error">Please check your username and password.</string>
     <string name="error_network_connection">There was a connection problem. Please check your network settings and try again.</string>
     <string name="remember_me">Remember me</string>
     <string name="umobile_header">uMobile Header</string>

